### ISSUE TITLE
Android app doesn't redirect when I click on deep link if the app is close, but it does if the app is already opened in background

### ISSUE THREAD

**Issue by gls-labs** [BranchMetrics/cordova-ionic-phonegap-branch-deep-linking#437]
I have an ionic app running on iOS and Android. For iOS devices, when I click the link, it redirects correctly to the page I want to visualize, but in android when I click on the link and the app is not active or already opened, it doesn't redirect and stays at the home page of the app, in Android if the app is already opened in background, the redirection occurs very well.

Any ideas??

**Comment by ethanneff** [comment]
Hello,

Make sure you have the `Branch.initSession()` being called in both `deviceready` and `resume`. https://github.com/BranchMetrics/cordova-ionic-phonegap-branch-deep-linking#initialize-branch

This will make sure your deep link data will flow through on both backgrounded and termination states of your app on Android. 

Thanks,

**Comment by gls-labs** [comment]
Yeah, it is exactly as you described. Is there a way to implement this as a synchronous solution to avoid other function interrupt the flow (redirects), for example, before we installed branch, we had our application flow which defines the page to go considering the status of the user, now we have implemented branch but I am suspecting that the problem is that the application makes redirects before we have read data from branch links.

**Comment by ethanneff** [comment]
Unfortunately, the `Branch.initSession()` is an async method. A connection to the Branch servers must be made to fingerprint the link click and pass the correct link data into your app. 

I guess we could build out a synchronous solution where you can read the link url to determine the in-app redirection. However, this solution will only work if a user 1) has the app 2) and navigates into it via a uri scheme example://. This synchronous will not work if a user 1) needs to install the app first, or 2) opens the app with a Universal Link, or 3) opens the app with a chrome intent, or 4) opens the app with an App Link.

Branch uses multiple ways of getting the app to open. Unfortunately, most of these methods require an async call to fingerprint the user and link.
